salt/proxy/napalm.py

```diff
@@ -194,13 +194,18 @@ def init(opts):
     """
     Opens the connection with the network device.
     """
-    __context__["napalm_device"] = {
-        "opts": opts,
-        "id": opts["id"],
-        "network_device": salt.utils.napalm.get_device(opts),
-        "details": {"initialized": True},
-    }
-    return True
+    try:
+        napalm_device = salt.utils.napalm.get_device(opts)
+        __context__["napalm_device"] = {
+            "opts": opts,
+            "id": opts["id"],
+            "network_device": napalm_device,
+            "details": {"initialized": True},
+        }
+        return True
+    except Exception as error:  # pylint: disable=broad-except
```

Comment:
I think we shouldn't catch the generic exception here, as we want to have the proxy attempt to re-connect if it fails first time (otherwise it'll just won't provide the `napalm_device` key in the context). Think for example of a device that isn't reachable when the Proxy Minion starts up, but it may be reachable later: in the form proposed here, the PM starts like everything would be fine, but in reality it's actually useless. Or, if we hard fail, but PM will re-start and reattempt to connect until we can probably connect to the device.